doc/development/api/cookieconsent.rst

```diff
@@ -0,0 +1,119 @@
+.. highlight:: python
+   :linenothreshold: 5
+
+.. _`cookieconsent`:
+
+Handling cookie consent
+=======================
+
+pretix includes an optional feature to handle cookie consent explicitly to comply with EU regulations.
+If your plugin sets non-essential cookies or includes a third-party service that does so, you should
+integrate with this feature.
+
+Server-side integration
+-----------------------
+
+First, you need to declare that you are using non-essential cookies by responding to the following
+signal:
+
+.. automodule:: pretix.presale.signals
+   :members: register_cookie_providers
+
+You are expected to return a list of ``CookieProvider`` objects instantiated from the following class:
+
+.. class:: pretix.presale.cookies.CookieProvider
+
+   .. py:attribute:: CookieProvider.identifier
+
+      A short and unique identifier used to distinguish this cookie provider form others (required).
+
+   .. py:attribute:: CookieProvider.provider_name
+
+      A human-readable name of the entity of feature responsible for setting the cookie (required).
+
+   .. py:attribute:: CookieProvider.usage_classes
+
+      A list of enum values from the ``pretix.presale.cookies.UsageClass`` enumeration class, such as
+      ``UsageClass.ANALYTICS``, ``UsageClass.MARKETING``, or ``UsageClass.SOCIAL`` (required).
+
+   .. py:attribute:: CookieProvider.privacy_url
+
+      A link to a privacy policy (optional).
+
+Here is an example of such a receiver:
+
+.. code-block:: python
+
+   @receiver(register_cookie_providers)
+   def recv_cookie_providers(sender, request, **kwargs):
+       return [
+           CookieProvider(
+               identifier='google_analytics',
+               provider_name='Google Analytics',
+               usage_classes=[UsageClass.ANALYTICS],
+           )
+       ]
+
+JavaScript-side integration
+---------------------------
+
+The server-side integration only causes the cookie provider to show up in the cookie dialog. You still
+need to care about actually enforcing the consent state.
+
+You can access the consent state through the ``window.pretix.cookie_consent`` variable. Whenever the
+value changes, a ``pretix:cookie-consent:change`` event is fired on the ``document`` object.
+
+The variable will generally have one of the following states:
+
+.. rst-class:: rest-resource-table
+
+===================================================== =====================================================
+State                                                 Interpretation
+===================================================== =====================================================
+``pretix.cookie_consent === undefined``               Your JavaScript has loaded before the cookie consent
```

Comment:
Currently that is true, but it might happen sometime that we define `window.pretix` somewhere else and cookie_consent is loaded later. Should we include this in the docs so that both cases (either `window.pretix` is undefined or even `window.pretix.cookie_consent`) are documented? The JS-code examples already cater for this by using `(window.pretix || {}).cookie_consent`.

```suggestion
``pretix === undefined``                              Your JavaScript has loaded before the cookie consent
``pretix.cookie_consent === undefined``               Your JavaScript has loaded before the cookie consent
```